service/relayer/Dockerfile

```
@@ -1,28 +1,51 @@
-WORKDIR /usr/src/relayer
+FROM debian:bookworm-slim AS node-dist
+ARG NODE_VERSION
+ARG NODE_DIST
+ARG NODE_SHA256
+
+# Fetch Node once, verify it, then reuse it in later stages.
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends ca-certificates curl xz-utils && \
+    rm -rf /var/lib/apt/lists/*
+
+RUN curl -fsSL "https://nodejs.org/dist/v${NODE_VERSION}/${NODE_DIST}.tar.xz" -o /tmp/node.tar.xz && \
```

Hadolint (2.14.0)

[warning] Set the SHELL option -o pipefail before RUN with a pipe in it. If you are using /bin/sh in an alpine image or if your shell is symlinked to busybox then consider explicitly setting your SHELL to /bin/ash, or disable this check

(DL4006)